client/src/components/App.jsx

```diff
@@ -100,9 +107,20 @@ class App extends React.Component {
       newPosition = position + 1;
     }
 
+    this.slideEffect(newPosition)
+  }
+
+  slideEffect(position) {
     this.setState({
-      position: newPosition
+      sliding: true,
+      position: position
     })
+    setTimeout(() => {
+
+      this.setState({
+        sliding: false,
+      })
+    }, 50);
```

Comment:
Is the setTimeout to ensure the first setState occurred? If yes, you can pass the setTimeout function as a callback to the first setState (so you are certain it occurred). 

this.setState({stuff:toChange}, ()=>{second setState here})